common/payload/payload.go

```diff
@@ -86,10 +86,13 @@ func MergeMapOfPayload(
 	m1 map[string]*commonpb.Payload,
 	m2 map[string]*commonpb.Payload,
 ) map[string]*commonpb.Payload {
-	if m1 == nil {
-		return maps.Clone(m2)
+	if m2 == nil {
+		return maps.Clone(m1)
```

Comment:
Why not just `return m1` ?

Reply:
Well, I think I could. But thought about what if it's changed outside after this function is called. It would change the map reference, and have unexpected side effects maybe.